processing/src/main/java/io/druid/segment/SimpleQueryableIndex.java

```diff
@@ -137,9 +142,28 @@ public Metadata getMetadata()
   private void initDimensionHandlers()
   {
     for (String dim : availableDimensions) {
+      if (Strings.isNullOrEmpty(dim)) {
+        continue;
+      }
       ColumnCapabilities capabilities = getColumn(dim).getCapabilities();
       DimensionHandler handler = DimensionHandlerUtils.getHandlerFromCapabilities(dim, capabilities, null);
       dimensionHandlers.put(dim, handler);
     }
   }
+  
+   /**
+    * Populate available dimensions, removes null/empty dimensions
+   */
+  private void removeNullorEmptyDimensions() 
+  {
+    String[] dimArr = new String[dimensionHandlers.keySet().size()];
+    int index = 0;
+    for (String dim : availableDimensions) {
```

Comment:
Better
```java
String[] dims = availableDimensions
    .stream()
    .filter(d -> !Strings.isNullOrEmpty(d))
    .toArray(String[]::new);
return new ArrayIndexed<>(dims, String.class);
```

But, again, I'm not sure that this is the right thing to do ultimately